pkg/icingadb/ha.go

```diff
@@ -228,6 +239,14 @@ func (h *HA) controller() {
 				h.signalHandover("lost heartbeat")
 				h.realizeLostHeartbeat()
 			}
+
+			// Reset retry timeout so that the next iterations have the full amount of time available again.
+			// Don't be surprised by the location of the code,
+			// as it is obvious that the timer is also reset after an error that ends the loop anyway.
+			// But this is the best place to catch all scenarios where the timeout needs to be reset.
+			// And since HA needs quite a bit of refactoring anyway to e.g. return immediately after calling h.abort(),
+			// it's fine to have it here for now.
+			retryTimeout.Reset(retry.DefaultTimeout)
```

Comment:
[`Timer#Reset()`](https://pkg.go.dev/time#Timer.Reset) emphasises that this receiver should only be called on an expired or stopped timer with drained channels, which is not the case here. Did you take these remarks into account when doing this here?

Apart from that, shouldn't the timeout be reset when entering this [code path](https://github.com/Icinga/icingadb/pull/698/files#diff-c07d29d7c4c9e56097cbdd2d4aba0dd00b2e13deac34ce4ab96d80ed63bbe821R176-R179) as well? Otherwise Icinga DB would simply crash after the timer was fired just because it *received a heartbeat from the past*.

Reply:
On first glance, it seems like the `time.Timer` can be replaced by a `time.Ticker`, which seems to be "reset-safe": <https://pkg.go.dev/time#Ticker.Reset>.

Reply:
> Apart from that, shouldn't the timeout be reset when entering this [code path](https://github.com/Icinga/icingadb/pull/698/files#diff-c07d29d7c4c9e56097cbdd2d4aba0dd00b2e13deac34ce4ab96d80ed63bbe821R176-R179) as well? Otherwise Icinga DB would simply crash after the timer was fired just because it _received a heartbeat from the past_.

What about this part of my previous comment? Doesn't the ticker have to be reset in this case too?

Reply:
@oxzi This is the place where the timeout **must not** be reset.
@yhabteab You're right.